src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -38,7 +38,7 @@ public static void init() throws URISyntaxException, RelayException {
 	@AfterClass
 	public static void cleanup() {
```

Comment:
>cleanup [](start = 20, length = 7)

It _really_ bothers me that there's not a single test case which closes a listener. We could be missing a bunch of shutdown issues.

Reply:
I'd be quite tempted to have each test case start/stop their own listener and have many/most of those tests fail if stopping throws any exception (e.g. no catch block around listener.close).

---
In reply to: [260847718](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260847718) [](ancestors = 260847718)

Reply:
Or maybe there could be another class with some but not all of these test cases which does open/close on a per-test basis?

---
In reply to: [260848343](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260848343) [](ancestors = 260848343,260847718)

Reply:
I agree, I will try to come up with some tests for closeAsync() in another PR

---
In reply to: [260849007](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260849007) [](ancestors = 260849007,260848343,260847718)